### ISSUE TITLE
View Base64 encoded images in DBeaver

### ISSUE THREAD

**Issue by tscrosh** [dbeaver/dbeaver#18498]
Hi, maybe it exists already but I can't find it
I have encode images in BASE64 and stored them on SQLSERVER - Table in a varbinary field.
Should be nice to have a Base64 decoder / image viewer, so we can see (may be manipulate/edit) the image directly in DBeaver
Thanks

**Comment by ShadelessFox** [comment]
Thank you for your request.

We will add the Base64 transformer.

**Comment by Torbins** [comment]
I hope to see Base64 transformer soon.

**Comment by CasualDeveloper** [comment]
A Base64 transformer would be really useful!

There's a program my work uses that stores SOAP message payloads as base64 encoded strings in a DB table and it would be much more convenient if DBeaver can just decode that automatically.

Currently I have to use another program to batch Base64 decode the column values before it's usable for me

**Comment by E1izabeth** [comment]
@MichalFoksa 
No one is working on it. If you want to contribute, you're welcome

**Comment by CaduVarela** [comment]
I'm thinking about implementing this as a value viewer with on-demand decode (no automatic detection). Does that approach make sense?

**Comment by Torbins** [comment]
I do not see any problem in decoding it automatically, if blob size is below 500 kb. It should be very fast.

**Comment by samanthaross926-pixel** [comment]
I see no problem at my end

**Comment by ShadelessFox** [comment]
@CaduVarela you can work on this issue. Please see our contribution guide first: https://github.com/dbeaver/dbeaver/wiki/Contribute-your-code

**Comment by seroperson** [comment]
@CaduVarela as a result, are you picking up this? I can give it a try if not.

**Comment by CaduVarela** [comment]
@seroperson 
Yes, I'm planning to work on it, probably starting today or tomorrow. That said, feel free to take it if you want to start earlier, no problem.

**Comment by seroperson** [comment]
@CaduVarela oh, yea, okay, good luck with implementation 👍 I will find something else to not duplicate the work.

**Comment by samanthaross926-pixel** [comment]
Thanks for taking this on. I look forward to seeing the implementation!
